### ISSUE TITLE
How do I read in Machine Comprehension predict output?

### ISSUE THREAD

**Issue by dhairyadalal** [allenai/allennlp#1172]
I'm a bit confused as to the format that returned by running the predict command and how to programmatically access it. 

For example, when I run the following example code from site: 

```
echo '{"passage": "A reusable launch system (RLS, or reusable launch vehicle, RLV) is a launch system which is capable of launching a payload into space more than once. ", "question": "What is RLS"}' > examples.jsonl

python -m allennlp.run predict \
    https://s3-us-west-2.amazonaws.com/allennlp/models/bidaf-model-2017.09.15-charpad.tar.gz \
    examples.jsonl > output.txt
```

I get back a file formatted like:
```
(allen_nlp) MacBook-Pro-5:AllenNLP dhairya$ cat output.txt 
input:  {'passage': 'A reusable launch system (RLS, or reusable launch vehicle, RLV) is a launch system which is capable of launching a payload into space more than once. ', 'question': 'What is RLS'}
prediction:  {"passage_question_attention": [[0.6167629361152649, 0.2354782521724701, 0.1477588564157486], [0.6327872276306152, 0.25898393988609314, 0.10822883248329163], [0.5114424228668213, 0.2422594428062439, 0.24629813432693481], [0.5788295269012451, 0.29724520444869995, 0.12392525374889374], [0.35226598381996155, 
```

From the documentation, it seem you're providing a HOCON formatted file. But I have issues reading in using pyhocon. I tried reading this like so: 
```
import json
from pyhocon import ConfigFactory

cf = ConfigFactory.parse_file("output.txt")
```
But that crashes with the following error: 
ParseSyntaxException: Expected "}" (at char 9), (line:1, col:10)

What exactly is the format of the output returned by the predict command? Also I'd appreciate any tips on how to read it in programmatically. 

Thanks!

**Comment by joelgrus** [comment]
two things here:

(1) you probably don't want to pipe the output to a file (because it has the `input:` and `prediction:`) prefixes. instead just use the `--output-file` parameter

(2) if you use `output-file` the results should be one JSON blob per line, so you can just do 

```
with open('output.txt') as f:
    for line in f:
        result = json.loads(line)
        # do something with the result
```

BUT if you really want to do programmatic predictions and use the output in Python, you might just want to do that directly. look at https://github.com/allenai/allennlp/blob/master/tests/service/predictors/bidaf_test.py#L11

basically you would need to

* use `load_archive` to get a model out of the serialized model
* call `Predictor.from_archive` to get the predictor
* make calls to `predictor.predict` (the unit test calls `predict_json`, which you could do too, but `predict` is maybe cleaner, see https://github.com/allenai/allennlp/blob/master/allennlp/service/predictors/bidaf.py#L15

**Comment by dhairyadalal** [comment]
thanks!
